src/table/internal.tsx

```diff
@@ -271,11 +274,14 @@ const InternalTable = React.forwardRef(
           __disableFooterDivider={true}
           __disableStickyMobile={false}
           footer={
-            footer && (
+            hasFooter ? (
               <div className={clsx(styles['footer-wrapper'], styles[`variant-${computedVariant}`])}>
-                <div className={styles.footer}>{footer}</div>
+                <div className={clsx(styles.footer, hasMobilePagination && styles['footer-with-pagination'])}>
+                  {footer && <span>{footer}</span>}
+                  {hasMobilePagination && <div className={styles['footer-pagination']}>{pagination}</div>}
```

Comment:
Typically, `footer` slot contains center-aligned text. How does this line up when pagination adds? Could you share the design specs in DM?